### DISCUSSION TITLE
How to set generated exe file icon correctly?

### DISCUSSION THREAD

**Discussion by garawaa** [pyinstaller/pyinstaller#8584] [Help]
I tried something like `pyinstaller -F -w --icon=icon.ico main.py` in my project directory. it does not change exe's icon? what is wrong?

**Comment by rokm** [comment]
That should work. Can you provide an example icon file that doesn't work? What version of Windows are you using?

**Reply by rokm** [reply]
Hmm, nevermind, it doesn't seem to work for me, either.

**Reply by rokm** [reply]
Actually, this looks like some sort of caching issue on Windows' side. 

If I build an executable without icon set, I get the default icon, which is expected. If I rebuild with icon set, the executable still has the default icon shown. If I rename the executable, the icon changes to the one that was set. If I rename the executable back to original name, the icon goes back to default. Copying the executable to a different location also updates the icon.
